File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        
        // 신고 당한 결과 (신고 당한 사용자 -> 해당 사용자를 신고한 사용자 목록)
        Map<String, Set<String>> reportedMap = new HashMap<>();
        
        // 신고한 결과 (신고한 사용자 -> 해당 사용자가 신고한 사용자 목록)
        Map<String, Set<String>> reportMap = new HashMap<>();
        
        for(String rp : report) {
            String to = rp.split(" ")[0];
            String from = rp.split(" ")[1];
            
            reportedMap.computeIfAbsent(from, value -> new HashSet<>()).add(to);
            reportMap.computeIfAbsent(to, value -> new HashSet<>()).add(from);
        }
        
        // 정지된 사용자
        List<String> suspendedUsers = new ArrayList<>();
        for(Map.Entry<String, Set<String>> entry : reportedMap.entrySet()) {
            String reportedUser = entry.getKey();
            int count = entry.getValue().size();
            
            if(count >= k) suspendedUsers.add(reportedUser);
        }
        
        // 각 회원 별 신고한 사용자가 정지당한 횟수 구하기
        int[] answer = new int[id_list.length];
        int idx = 0;
        for(String id : id_list) {
            Set<String> reportedByUsers = reportMap.get(id);
            
            int cnt = reportedByUsers != null ? (int) reportedByUsers.stream()
                .filter(suspendedUsers::contains)
                .count() : 0;
            
            answer[idx++] = cnt;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.kt
class Solution {
    fun solution(id_list: Array<String>, report: Array<String>, k: Int): IntArray {
        
        // 신고결과
        val reportResultMap: MutableMap<String, MutableList<String>> = mutableMapOf()
        
        // 신고횟수
        val reportCountMap: MutableMap<String, Int> = mutableMapOf()
        
        for(rp in report) {
            val (to, from) = rp.split(" ")
            
            // 신고결과 Map에 신고한 유저의 id가 없는 경우, key 생성 및 MutableList 초기화
            val list = reportResultMap.getOrPut(to) { mutableListOf() }
            
            // 신고당한 회원이 존재하지 않는 경우 (중복 신고 방지)
            if(!list.contains(from)) {
                list.add(from)
                reportCountMap[from] = reportCountMap.getOrDefault(from, 0) + 1
            }
        }
        
        val answer = IntArray(id_list.size)
        for((idx, id) in id_list.withIndex()) {
            var count = 0
            
            // 유저의 신고 결과 목록을 순회하면서 k번 이상 신고당했을 경우 처리 결과 횟수 + 1
            for(reportedUser in reportResultMap[id] ?: emptyList()) {
                if((reportCountMap[reportedUser] ?: 0) >= k) count += 1
            }
            
            answer[idx] = count
        }
        
        return answer
    }
}